presto-function-namespace-managers/src/main/java/com/facebook/presto/functionNamespace/json/JsonFileBasedFunctionDefinitionProvider.java

```diff
@@ -28,11 +38,35 @@
     public UdfFunctionSignatureMap getUdfDefinition(String filePath)
     {
         try {
-            return parseJson(Paths.get(filePath), UdfFunctionSignatureMap.class);
+            Map<String, List<JsonBasedUdfFunctionMetadata>> udfSignatureMap = new HashMap<>();
+            for (Path path : getFilesInPath(filePath)) {
+                // Note: Latest updates take effect in case of duplicates.
+                udfSignatureMap.putAll(parseJson(path, UdfFunctionSignatureMap.class).getUDFSignatureMap());
+            }
+            return new UdfFunctionSignatureMap(udfSignatureMap);
         }
         catch (Exception e) {
             log.info("Failed to load function definition for JsonFileBasedFunctionNamespaceManager " + e.getMessage());
         }
         return null;
     }
+
+    // Returns the list containing the file if filePath is a single file or all json files inside filePath if it's a dir
+    private List<Path> getFilesInPath(String filePath) throws IOException
+    {
+        final String jsonFileExtension = ".json";
+        Path path = Paths.get(filePath);
+
+        if (Files.isDirectory(path)) {
+            try (Stream<Path> stream = Files.list(path)) {
+                return stream
+                        .filter(file -> !Files.isDirectory(file))
+                        .filter(file -> file.getFileName().toString().toLowerCase(ENGLISH).endsWith(jsonFileExtension))
+                        .collect(Collectors.toList());
+            }
+        }
+        else {
+            return Collections.singletonList(path);
```

Comment:
we generally use ImmutableList.of() rather than singletonList()

Reply:
Good point; will update! :)